specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscNode.json

```diff
@@ -0,0 +1,957 @@
+        },
+        "resourceId": {
+          "type": "string",
+          "description": "Gets or sets the resource ID which generated the error."
+        },
+        "errorCode": {
+          "type": "string",
+          "description": "Gets or sets the error code."
+        },
+        "errorMessage": {
+          "type": "string",
+          "description": "Gets or sets the error message."
+        },
+        "locale": {
+          "type": "string",
+          "description": "Gets or sets the locale of the error."
+        },
+        "errorDetails": {
+          "type": "string",
+          "description": "Gets or sets the error details."
+        }
+      },
+      "description": "Definition of the dsc node report error type."
+    },
+    "AgentRegistration": {
+      "properties": {
+        "dscMetaConfiguration": {
+          "type": "string",
+          "description": "Gets or sets the dsc meta configuration."
+        },
+        "endpoint": {
+          "type": "string",
+          "description": "Gets or sets the dsc server endpoint."
+        },
+        "keys": {
+          "$ref": "#/definitions/AgentRegistrationKeys",
+          "description": "Gets or sets the agent registration keys."
+        },
+        "id": {
+          "type": "string",
+          "description": "Gets or sets the id."
+        }
+      },
+      "description": "Definition of the agent registration infomration type."
+    },
+    "AgentRegistrationKeys": {
+      "properties": {
+        "primary": {
+          "type": "string",
+          "description": "Gets or sets the primary key."
+        },
+        "secondary": {
+          "type": "string",
+          "description": "Gets or sets the secondary key."
+        }
+      },
+      "description": "Definition of the agent registration keys."
+    },
+    "DscNode":{
+      "description": "Definition of a DscNode",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DscNodeProperties",
+          "x-ms-client-flatten": true,
+          "description": "The properties of a DscNode."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./definitions.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "DscNodeProperties" : {
+      "description": "The properties of a DscNode",
+      "properties": {
+        "lastSeen": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the last seen time of the node."
+        },
+        "registrationTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the registration time of the node."
+        },
+        "ip": {
+          "type": "string",
+          "description": "Gets or sets the ip of the node."
+        },
+        "accountId": {
+          "type": "string",
+          "description": "Gets or sets the account id of the node."
+        },
+        "nodeConfiguration": {
```

Comment:
>nodeConfiguration [](start = 9, length = 17)

WDYT about flattening this property?

Reply:
Could you describe a little more about this? You mean the x-ms-client-flatten property?

Reply:
yes

Reply:
Done.

Reply:
Fixed